db/collab.go

```diff
@@ -107,19 +119,27 @@ func (d *DB) JoinCollab(c echo.Context) error {
 
 		for {
 			var msg Message
-			err := websocket.JSON.Receive(ws, &msg)
-			if err != nil {
+
+			if err := websocket.JSON.Receive(ws, &msg); err != nil {
 				errorMsg := Message{
 					Author: uid,
 					Type:   "ERROR",
 					Body:   err.Error(),
 				}
-				session.Broadcast(errorMsg)
-				session.RemoveConn(uid)
+				if broadcastErr := session.Broadcast(errorMsg); broadcastErr != nil {
+					break
+				}
+
+				if removeErr := session.RemoveConn(uid); removeErr != nil {
+					break
+				}
+
 				break
 			}
```

Comment:
We'll probably have to clean this up a bit once we add actual functionality for processing specific messages, but this will be our NOP starter :+1: